### ISSUE TITLE
NVDA is not reading checkbox from header table in Firefox browser

### ISSUE THREAD

**Issue by MariaBulboaca** [nvaccess/nvda#14606]
### Steps to reproduce:

1. Open Firefox browser
2. Go to: [https://codepen.io/Maria-Bulboaca/pen/oNMQwow](https://codepen.io/Maria-Bulboaca/pen/oNMQwow)
3. Press tab until you focus the checkbox

### Actual behavior:

When the checkbox is focused the NVDA does not read anything

### Expected behavior:

When the checkbox is focused the NVDA should read 'Select all checkbox not checked'

### NVDA logs, crash dumps and other attachments:
I have attached here an image of the speech viewer when the checkbox is focused.
![image](https://user-images.githubusercontent.com/108864526/216289237-7c5db080-4b22-47a6-aeb5-4679973026f3.png)

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
NVDA 2022.2

#### Windows version:
Windows 10 Enterprise

#### Name and version of other software in use when reproducing the issue:

None
I tested with other screen readers(JAWS 2023 and Narrator) and it works.

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, NVDA 2022.4 and the NVDA behave the same.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by CyrilleB79** [comment]
Hi @MariaBulboaca 

Welcome to NVDA bugtracker. And thanks for having followed the template.

Could you please keep in mind that some developers of NVDA are blind. Thus it's better to provide a text description than an image, especially when the information to be communicated is text. In your case, could you please edit your initial description replacing the log image with a file (if needed)? Many thanks!

**Comment by MariaBulboaca** [comment]
Hi, @CyrilleB79

I added a short description of the image because there was no need for a file.

**Comment by MariaBulboaca** [comment]
Small update: I discovered today that the same issue is happening and for title cell header(a simple text). If the table has vertical or horizontal scroll, the NVDA will not read the first cell in the header after the table is focused.

**Comment by jcsteh** [comment]
This occurs because the table is focusable and the check box is the very first element in the table. Thus, the start of the table and the check box are the same point for the browse mode cursor. When you tab to the table, the browse mode cursor moves to that spot. When you tab to the check box, the browse mode cursor doesn't move because it's already there, so NVDA reports nothing, even though the actual focus changed.

@michaelDCurran, I'm guessing this might be fixed by #14611?

**Comment by jcsteh** [comment]
Btw, the reason this doesn't occur in Chrome right now is that scrollable areas aren't focusable by default in Chrome. That is likely to change in the future, however, at which point you will see the same behaviour.

**Comment by michaelDCurran** [comment]
Yes should be fixed by #14611
